students/BrianNisonger/session9/mailroom.py

```diff
@@ -0,0 +1,197 @@
+        self.donor_list = donor_list
+
+    def append(self, cls):
+        """Adds another donor to the donor_list"""
+        self.donor_list.append(cls)
+
+    def list_donors(self):
+        """returns a list containing the names of all the donors in the donor list"""
+        return [donor.name for donor in self.donor_list]
+
+    def make_headers_string(self):
+        """Header for the report"""
+        headers = ("Donor Names", "Total Given", "Num Gifts", "Average Gifts")
+        header_string = f'{headers[0]:<20} {headers[1]:<15} {headers[2]:<15} {headers[3]:<15}'
+        return header_string
+
+    def sort_donors(self):
+        """Takes a list of donors and sorts by total donations"""
+        self.donor_list = sorted(self.donor_list)
+
+    def make_report(self):
+        """Combines the report header with individual report strings"""
+        report = self.make_headers_string() + "\n"
+        self.sort_donors()
+        for donor in self.donor_list:
+            report += donor.make_report_string(donor.donation) + "\n"
+        return report
+
+    def make_thank_notes(self):
+        """Provides a list containing all the thank you notes for Donors, Not Implemented in final code"""
+        self.thank_notes = []
+        for donor in self.donor_list:
+            self.thank_notes.append(donor.make_thank_you())
+        return self.thank_notes
+
+    def find_donor(self, value):
+        """Takes a string value and returns a donor class if donor name matches string, or a empty list if not"""
+        donor_class = []
+        for donor in self.donor_list:
+            if donor.name.lower() == value.lower():
+                donor_class = donor
+        return donor_class
+
+    def make_filenames(self):
+        """Makes a list of filenames for the donor list, Not implemented in final code"""
+        filenames = []
+        for donor in self.donor_list:
+            filenames.append(donor.make_filename())
+        return filenames
+
+
+def send_file(msg, filename):
+    """Take a string and a filename and write the msg out to the file specified"""
+    with open(filename, "w") as outfile:
+        outfile.write(msg)
+
+
+def add_donation(d_list, full_name):
+    """Takes the list of donor classes and a full name and adds a donation using user input"""
+    donor = d_list.find_donor(full_name)
+    if type(donor) == list:
+        donor = Donor(full_name)
+        d_list.append(donor)
+    donation = input("Please enter a donation==>")
+    while not donation.isdigit():
+        donation = input("Please enter a donation==>")
+    donation = float(donation)
+    donor.add_donation(donation)
+    send_file(donor.make_thank_you(), donor.make_filename())
+
+
+def thank_you(d_list):
+    """Gnerates a thank you for a new donation, using user input it either lists the donors, or otherwise adds a donation to the list"""
+    while True:
+        full_name = input("Please enter a Full Name==>")
+        if full_name.lower() == "list":
```

Comment:
this logic should be in the Donors class -- not here! This code shoulnd't know how the Donors are stored, nor how to find them.

don't you have a method for this in your Donors class?